### ISSUE TITLE
Update solidity pragma directive to ^0.7.0

### ISSUE THREAD

**Issue by tab00** [Uniswap/uniswap-v2-core#108]
Please upgrade contracts to use the latest version.

See which contracts need to be updated simply by searching: https://github.com/Uniswap/uniswap-v2-core/search?q=pragma

0.7.0 was released in July 2020 and the latest version is 0.7.2. 
0.5.0 was released almost 2y ago.
See the full solc release list at http://solc-bin.ethereum.org/bin/list.json .

Some contracts specify an exact solidity version, which is problematic as they won't compile. e.g. a contract having `pragma solidity =0.5.16;` would not compile if solc version 0.5.17 is used.

**Comment by sujithsomraaj** [comment]
> Please upgrade contracts to use the latest version.
> 
> See which contracts need to be updated simply by searching: https://github.com/Uniswap/uniswap-v2-core/search?q=pragma
> 
> 0.7.0 was released in July 2020 and the latest version is 0.7.2.
> 0.5.0 was released almost 2y ago.
> See the full solc release list at http://solc-bin.ethereum.org/bin/list.json .
> 
> Some contracts specify an exact solidity version, which is problematic as they won't compile. e.g. a contract having `pragma solidity =0.5.16;` would not compile if solc version 0.5.17 is used.

Yes, i too have issues in using V2 router cause certain contracts have compiler version strictly set to =0.5.16; and =0.6.6;
